go/test/endtoend/sharding/verticalsplit/vertical_split_test.go

```diff
@@ -0,0 +1,766 @@
+	moving2FirstAdd1 := insertValues(t, conn, sourceKeyspace, "master", "moving2", 100)
+	checkValuesTimeout(t, destinationMasterTablet, "vt_destination_keyspace", "moving1", moving1FirstAdd1, 100, 30)
+	checkValuesTimeout(t, destinationMasterTablet, "vt_destination_keyspace", "moving2", moving2FirstAdd1, 100, 30)
+	sharding.CheckBinlogPlayerVars(t, destinationMasterTablet, []string{sourceKs}, 30)
+	sharding.CheckBinlogServerVars(t, sourceReplicaTablet, 100, 100, true)
+
+	// use vtworker to compare the data
+	t.Log("Running vtworker VerticalSplitDiff")
+	err = clusterInstance.VtworkerProcess.ExecuteVtworkerCommand(clusterInstance.GetAndReservePort(),
+		clusterInstance.GetAndReservePort(),
+		"--use_v3_resharding_mode=true",
+		"--cell", "test_nj",
+		"VerticalSplitDiff",
+		"--min_healthy_rdonly_tablets", "1",
+		"destination_keyspace/0")
+	assert.Nil(t, err)
+
+	// get status for destination master tablet, make sure we have it all
+	sharding.CheckRunningBinlogPlayer(t, destinationMasterTablet, 700, 300)
+
+	// check query service is off on destination master, as filtered
+	// replication is enabled. Even health check should not interfere.
+	destinationMasterTabletVars := destinationMasterTablet.VttabletProcess.GetVars()
+	assert.NotNil(t, destinationMasterTabletVars)
+	assert.Contains(t, reflect.ValueOf(destinationMasterTabletVars["TabletStateName"]).String(), "NOT_SERVING")
+
+	// check we can't migrate the master just yet
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("MigrateServedFrom", "destination_keyspace/0", "master")
+	assert.NotNil(t, err)
+
+	// migrate rdonly only in test_ny cell, make sure nothing is migrated
+	// in test_nj
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("MigrateServedFrom", "--cells=test_ny", "destination_keyspace/0", "rdonly")
+	assert.Nil(t, err)
+
+	// check SrvKeyspace
+	checkSrvKeyspaceServedFrom(t, cellj, destinationKeyspace, ksServedFrom, *clusterInstance)
+	checkBlacklistedTables(t, sourceMasterTablet, sourceKeyspace, nil)
+	checkBlacklistedTables(t, sourceReplicaTablet, sourceKeyspace, nil)
+	checkBlacklistedTables(t, sourceRdOnlyTablet1, sourceKeyspace, nil)
+	checkBlacklistedTables(t, sourceRdOnlyTablet2, sourceKeyspace, nil)
+
+	// migrate test_nj only, using command line manual fix command,
+	// and restore it back.
+	keyspaceJSON, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("GetKeyspace", "destination_keyspace")
+	assert.Nil(t, err)
+
+	var keyspace topodata.Keyspace
+	err = json2.Unmarshal([]byte(keyspaceJSON), &keyspace)
+	assert.Nil(t, err)
+	found := false
+	for _, servedFrom := range keyspace.GetServedFroms() {
+		if strings.ToLower(servedFrom.GetTabletType().String()) == "rdonly" {
+			found = true
+			assert.Contains(t, strings.Join(servedFrom.GetCells(), " "), "test_ca")
+			assert.Contains(t, strings.Join(servedFrom.GetCells(), " "), "test_nj")
+		}
+	}
+	assert.Equal(t, true, found)
```

Comment:
Maybe easier to review if moved to a separate private method.

Reply:
Done. Refer [PR46](https://github.com/planetscale/vitess/pull/46)